modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -60,28 +60,34 @@ namespace cv
 //! @addtogroup imgcodecs
 //! @{
 
-enum { IMREAD_UNCHANGED  = -1, // 8bit, color or not
-       IMREAD_GRAYSCALE  = 0,  // 8bit, gray
-       IMREAD_COLOR      = 1,  // ?, color
-       IMREAD_ANYDEPTH   = 2,  // any depth, ?
-       IMREAD_ANYCOLOR   = 4,  // ?, any color
-       IMREAD_LOAD_GDAL  = 8   // Use gdal driver
+//! Imread flags
+enum ImreadModes {
+       IMREAD_UNCHANGED  = -1, //!< If set, return the loaded image as is (with alpha channel, otherwise it gets cropped)
+       IMREAD_GRAYSCALE  = 0,  //!< If set, always convert image to the single channel grayscale image
+       IMREAD_COLOR      = 1,  //!< If set, always convert image to the 3 channel BGR color image
+       IMREAD_ANYDEPTH   = 2,  //!< If set, return 16-bit/32-bit image when the input has the corresponding depth, otherwise convert it to 8-bit.
+       IMREAD_ANYCOLOR   = 4,  //!< If set, the image is read in any possible color format
+       IMREAD_LOAD_GDAL  = 8   //!< If set, use the gdal driver for loading the image
      };
 
-enum { IMWRITE_JPEG_QUALITY        = 1,
-       IMWRITE_JPEG_PROGRESSIVE    = 2,
-       IMWRITE_JPEG_OPTIMIZE       = 3,
-       IMWRITE_JPEG_RST_INTERVAL   = 4,
-       IMWRITE_JPEG_LUMA_QUALITY   = 5,
-       IMWRITE_JPEG_CHROMA_QUALITY = 6,
-       IMWRITE_PNG_COMPRESSION     = 16,
-       IMWRITE_PNG_STRATEGY        = 17,
-       IMWRITE_PNG_BILEVEL         = 18,
-       IMWRITE_PXM_BINARY          = 32,
-       IMWRITE_WEBP_QUALITY        = 64
+//! Imwrite flags
+enum ImwriteFlags {
```

Comment:
- IMWRITE_JPEG_PROGRESSIVE, IMWRITE_JPEG_OPTIMIZE - enable JPEG features, 0 or 1, default is False
- IMWRITE_JPEG_RST_INTERVAL - restart interval, 0 - 65535, default is 0 - no restart
- IMWRITE_JPEG_LUMA_QUALITY, IMWRITE_JPEG_CHROMA_QUALITY - separate quality levels, 0 - 100, default is 0 - don't use
- IMWRITE_PNG_STRATEGY - one of cv::ImwritePNGFlags, default is IMWRITE_PNG_STRATEGY_DEFAULT
- IMWRITE_PNG_BILEVEL - binary level PNG, 0 or 1, default is 0